pkg/controller/daemon/daemoncontroller.go

```diff
@@ -89,6 +89,9 @@ type DaemonSetsController struct {
 	// Added as a member to the struct to allow injection for testing.
 	nodeStoreSynced cache.InformerSynced
 
+	// To allow injection of CreatePodTemplate for testing
```

Comment:
I don't think you need this. You can just use `kubeClient` and fake the pod template calls, right?